Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java

```diff
@@ -0,0 +1,34 @@
+package com.epam.gmail.test.driver;
+
+import org.openqa.selenium.WebDriver;
+
+public class WebDriverSingleton {
+
+	private static WebDriver driver;
+	
+	private static WebDriverCreator creator;
+
+	private WebDriverSingleton() {
+	}
+
+	public static void setCreator(WebDriverCreator newCreator) {
+		creator = newCreator;
+	}
+	
+	public static WebDriver getWebDriverInstance() {
+		if (driver == null) {
+			if(creator == null) {
+				// create chrome driver by default
+				driver = new ChromeDriverCreator().createWebDriver();
+			} else {
+				driver = creator.createWebDriver();
+			}
+			driver = new WebDriverDecorator(driver);
+		}
+		return driver;
+	}
+
+	public static void invalidateWebDriverInstance() {

```

Comment:
In my code I recreate driver instance to reopen browser to simplify change of users.